Source/VCGeneration/Checker.cs

```diff
@@ -184,26 +185,20 @@ private void SetRlimit(uint rlimit)
     {
       TheoremProver.SetRlimit(Util.BoundedMultiply(rlimit, 1000));
     }
-
-    private void SetRandomSeed(int? randomSeed)
-    {
-      TheoremProver.SetRandomSeed(randomSeed);
-    }
-
     
     /// <summary>
     /// Set up the context.
     /// </summary>
-    private void Setup(Program prog, ProverContext ctx, Split s = null)
+    private void Setup(Program prog, ProverContext ctx, Split split = null)
     {
-      var implementationRandomSeed = s?.Implementation.RandomSeed;
-      var random = implementationRandomSeed == null ? CommandLineOptions.Clo.Random : new Random(implementationRandomSeed.Value);
+      solverOptions.RandomSeed = split?.Implementation.RandomSeed ?? CommandLineOptions.Clo.RandomSeed;
```

Comment:
Here we are overwriting the class-level `solverOptions`; is that to allow setting per-procedure random seeds?  AFAICT this method doesn't otherwise mutate `this` (or does it?)

Reply:
> is that to allow setting per-procedure random seeds?

Yes, a checker is only used for one split at a time, and its settings get updated if a new split requires that.